File: cpp/303_区域和检索-数组不可变_easy.cpp
class NumArray {
private:
    vector<int> mm;
public:
    NumArray(vector<int>& nums) {
        mm.push_back(0);
        for (auto num : nums)
            mm.push_back(num+mm.back());
    }
    
    int sumRange(int i, int j) {
        return mm[j+1] - mm[i];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(i,j);
 */

File: python/303_区域和检索-数组不可变_easy.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.mm = [0]
        for num in nums:
            self.mm.append(num+self.mm[-1])

    def sumRange(self, i: int, j: int) -> int:
        return self.mm[j+1] - self.mm[i]
        



# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)